mm2src/crypto/src/encrypt.rs

```diff
@@ -0,0 +1,117 @@
+use crate::key_derivation::KeyDerivationDetails;
+use aes::cipher::{block_padding::Pkcs7, BlockEncryptMut, KeyIvInit};
+use aes::Aes256;
+use common::drop_mutability;
+use derive_more::Display;
+use hmac::{Hmac, Mac};
+use mm2_err_handle::prelude::*;
+use sha2::Sha256;
+
+type Aes256CbcEnc = cbc::Encryptor<Aes256>;
+
+#[derive(Debug, Display, PartialEq)]
+pub enum EncryptionError {
+    #[display(fmt = "Error generating random bytes: {}", _0)]
+    UnableToGenerateRandomBytes(String),
+    #[display(fmt = "AES cipher error: {}", _0)]
+    AESCipherError(String),
+    Internal(String),
+}
+
+/// Enum representing different encryption algorithms.
+#[derive(Serialize, Deserialize, Debug)]
+pub enum EncryptionAlgorithm {
+    /// AES-256-CBC algorithm.
+    AES256CBC,
+    // Placeholder for future algorithms.
+    // Future algorithms can be added here.
```

Comment:
These lines seems like duplicated, maybe it's just me, you can ignore this :)

Reply:
Done